### PR TITLE
Recursively camelCase binding trigger metadata

### LINKED ISSUE #39: recursive camelcase binding data

### PR DESCRIPTION
**Pull request by mhoeger**
Resolves #39

### CODE REVIEW

File: src/Context.ts [hunk 1]
```diff
@@ -106,22 +106,35 @@ function getLogger(invocationId: string, functionName: string, log: ILogCallback
...
   return bindingData;
 }
 
+// Recursively convert keys of objects to camel case
+function convertKeysToCamelCase(obj: any) {
+  var output = {};
+  for (var key in obj) {
+      let value = fromTypedData(obj[key]) || obj[key];
+      let camelCasedKey = key.charAt(0).toLocaleLowerCase() + key.slice(1);
+      // If the value is a JSON object, convert keys to camel case
```
Comment (pragnagopa) [inline src/Context.ts]: Can we add a unit test here? Verifying only via E2E tests is getting harder. We should try and improve unit test coverage for each language worker

**Comment by mhoeger** [comment]
@pragnagopa - added some initial tests! In running the tests, I realized that triggerMetadata includes things like the http object itself. Maybe this is desired behavior, but it could also be a bug on host?

File: test/ContextTests.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        }
+    };
+    var request: rpc.InvocationRequest$Properties = <rpc.InvocationRequest$Properties> {
+        triggerMetadata: triggerDataMock,
+        invocationId: "12341"
+    }
+    
+    var bindingData = getNormalizedBindingData(request);
+
+    expect(bindingData.invocationId).to.equal('12341');
```
Comment (pragnagopa) [inline test/ContextTests.ts]: Nit: Add comments //Verify conversion to camelCase

File: test/ContextTests.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    expect(bindingData.headers.connection).to.equal('Keep-Alive');
+    expect(bindingData.req.http.url).to.equal("https://mock");
+    expect(bindingData.sys.methodName).to.equal('test-js');
+    expect(bindingData.sys.utcNow).to.equal('2018');
+    expect(bindingData.sys.randGuid).to.equal('3212');
+
+    expect(bindingData.sys.UtcNow).to.be.undefined;
```
Comment (pragnagopa) [inline test/ContextTests.ts]: Nit: Similar comment here  //Verify accessing original keys is undefined

File: test/ContextTests.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+describe('Context', () => {
+  it('normalizes binding trigger metadata', () => {
+    var mockRequest: rpc.TypedData$Properties = toRpcHttp({ url: "https://mock"});
+    var triggerDataMock: { [k: string]: rpc.TypedData$Properties } = {
+        "Headers": {
+            json: JSON.stringify({Connection: 'Keep-Alive'})
+        },
+        "Req": mockRequest,
+        "Sys": {
+            json: JSON.stringify({MethodName: 'test-js', UtcNow: '2018', RandGuid: '3212'})
```
Comment (pragnagopa) [inline test/ContextTests.ts]: for completeness, add tests for other casing camelCased : 'testCamelCase', lowercased: 'testlowercase' etc

**Pull request merged**